iron-list.js

```diff
@@ -1323,10 +1323,17 @@ Polymer({
         }
       });
     } else {
+      let order = [];
       this._iterateItems(function(pidx, vidx) {
         this.translate3d(0, y + 'px', 0, this._physicalItems[pidx]);
         y += this._physicalSizes[pidx];
+        if (this._physicalItems[pidx].id) {
```

Comment:
Since this is in a hot path, can you capture `const item = this._physicalItems[pidx]` (used twice) and `item.id` (used twice) to cut down on the extra dereferences? 

Let's also add this code in for the grid case above; I think it should apply the same, right?

Reply:
Sounds good, added constants for item and itemID.

I'm not sure a grid would handle this the same way since a grid has both rows and columns. The ownership of the columns should be on the row, and the rows would be owned by the grid. I haven't looked into this enough to be confident that the fix for a list would be the same as a fix for a grid.

Reply:
I'm not an ARIA expert, but even in grid mode the data source is still a one-dimensional array, so the row-item association is purely a presentation concept since it depends on screen width. 

However, I just looked at the spec and apparently there are [`grid`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Grid_Role)-specific ARIA attributes, and I'd have to learn more to know whether that applies for `iron-list`'s type of grid or not (since we don't implement arrow-key 2D navigation, only tab).  So for now I agree we'll just treat that as a separate issue and go ahead and get the list-only support merged.